src/charts/stacked-bar.js

```diff
@@ -0,0 +1,1010 @@
+define(function(require){
+    'use strict';
+
+     const d3Array = require('d3-array'),
+     d3Axis = require('d3-axis'),
+     d3Color = require('d3-color'),
+     d3Collection = require('d3-collection'),
+     d3Dispatch = require('d3-dispatch'),
+     d3Ease = require('d3-ease'),
+     d3Scale = require('d3-scale'),
+     d3Shape = require('d3-shape'),
+     d3Selection = require('d3-selection'),
+    // const d3Transition = require('d3-transition');
+    // const d3TimeFormat = require('d3-time-format');
+
+     assign = require('lodash.assign'),
+     {exportChart} = require('./helpers/exportChart'),
+     colorHelper = require('./helpers/colors'),
+    // const timeAxisHelper = require('./helpers/timeAxis');
+     {isInteger} = require('./helpers/common'),
+     {axisTimeCombinations} = require('./helpers/constants'),
+     NUMBER_FORMAT = ',f',
+     {
+      formatIntegerValue,
+      formatDecimalValue,
+    } = require('./helpers/formatHelpers'),
+    uniq = (arrArg) => arrArg.filter((elem, pos, arr) => arr.indexOf(elem) == pos);
+
+
+    /**
+     * @typdef D3Layout
+     * @type function
+     */
+
+    /**
+     * @typedef stackedBarData
+     * @type {Object}
+     * @property {Object[]} data       All data entries
+     * @property {String} name         Name of the entry
+     * @property {Number} value        Value of the entry
+     *
+     * @example
+     * {
+     *     'data': [
+     *         {
+     *             "name": "2011-01",
+     *             "stack": "Direct",
+     *             "value": 0
+     *         }
+     *     ]
+     * }
+     */
+
+    /**
+     * Stacked Area Chart reusable API module that allows us
+     * rendering a multi area and configurable chart.
+     *
+     * @module Stacked-bar
+     * @tutorial stacked-bar
+     * @requires d3-array, d3-axis, d3-collection, d3-ease, d3-scale, d3-shape, d3-selection, d3-time, d3-time-format
+     *
+     * @example
+     * let stackedBar = stackedBar();
+     *
+     * stackedBar
+     *     .width(containerWidth);
+     *
+     * d3Selection.select('.css-selector')
+     *     .datum(dataset.data)
+     *     .call(stackedBar);
+     *
+     */
+
+     return function module() {
+
+        let margin = {
+            top: 70,
```

Comment:
This numbers feel a bit high for defaults, would it work with something between 20 and 40?